webapp/src/components/ImageFieldEditor.tsx

```diff
@@ -197,7 +226,82 @@ export class ImageFieldEditor<U extends pxt.Asset> extends React.Component<Image
         this.galleryAssets = getAssets(true, this.asset.type);
     }
 
-    protected filterAssets(assets: pxt.Asset[], type: pxt.AssetType = this.asset?.type, isGallery = false, useTags?: boolean) {
+    protected getAvailableTags(filterAssets: pxt.Asset[]) {
+        let collectedTags: string[] = [];
+
+        if (this.galleryAssets) {
+            const specialTags = ["tile", "dialog", "background"];
+            filterAssets.forEach( (asset) => {
+                if (asset.meta.tags) {
+                    asset.meta.tags.forEach(t => {
+                        let sanitizedTag = sanitize(t);
+                        if (specialTags.indexOf(sanitizedTag) == -1) {
+                            if (collectedTags.indexOf(sanitizedTag) < 0) {
+                                collectedTags.push(sanitizedTag);
+                            }
+                        }
+                    });
+                }
```

Comment:
```suggestion
                asset.meta.tags?.forEach(t => {
                    const sanitizedTag = sanitize(t);
                    if (specialTags.indexOf(sanitizedTag) == -1 && collectedTags.indexOf(sanitizedTag) < 0) {
                        collectedTags.push(sanitizedTag);
                    }
                });
```

nit, a lot of indentation here so maybe worth